content/docs/1.7.0/nodes-and-volumes/volumes/rwx-volumes.md

```diff
@@ -40,8 +40,65 @@ It is necessary to meet the following requirements in order to use RWX volumes.
-5. For a Longhorn volume that gets used by a RWX PVC, the volume access mode will be changed to RWX.
+5. For a Longhorn volume that gets used by an RWX PVC, the volume access mode will be changed to RWX.
 
+## Configuring Volume Locality for an RWX Volume
+
+  Longhorn introduces several enhancements to control the volume locality of RWX volumes. This is accomplished by determining the location of the share manager pod associated with an RWX volume. This section explains how to configure the data locality feature for RWX volumes.
+
+### Using `shareManagerNodeSelector`
+
+You can specify the `shareManagerNodeSelector` in the `storageClass.Parameters` to dictate on which nodes RWX volumes can be scheduled. These selectors are merged with global `system-managed-components-node-selector` settings and applied to the share manager pods of the RWX volumes. This provides more control over the locality of the RWX volumes.
+
+  Example:
+  ```
+  kind: StorageClass
+  apiVersion: storage.k8s.io/v1
+  metadata:
+    name: longhorn-rwx
+  provisioner: driver.longhorn.io
+  parameters:
+    shareManagerNodeSelector: label-key1:label-value1;label-key2:label-value2
+  ```
+  In the above example, RWX volumes provisioned with this StorageClass will be scheduled on nodes that have the label key=val. 
+
+### Using `allowedTopologies`
+
+Longhorn converts the `storageClass.allowedTopologies` settings into affinity rules for the share manager pods of the RWX volumes. This ensures the pods are scheduled on nodes that meet the specified topological requirements, such as regions or zones, aligning with the RWX volume locality.
+
+  Example:
+  ```
+  kind: StorageClass
+  apiVersion: storage.k8s.io/v1
+  metadata:
+    name: longhorn-rwx
+  provisioner: driver.longhorn.io
+  allowedTopologies:
+  - matchLabelExpressions:
+    - key: topology.kubernetes.io/region
+      values:
+      - us-west-1
+  ```
+  In the above example, the share manager pods and RWX volumes will be scheduled in the `us-west-1` region.
+
+### Using `shareManagerTolerations`
+
+You can define `shareManagerTolerations` in the `storageClass.Parameters`. These tolerations are merged with global `taint-toleration` settings and applied to the share manager pods, allowing more flexible scheduling based on node taints.
+
+  Example:
+  ```
+  kind: StorageClass
+  apiVersion: storage.k8s.io/v1
+  metadata:
+    name: longhorn-rwx
+  provisioner: driver.longhorn.io
+  parameters:
+    shareManagerTolerations: nodetype=storage:NoSchedule
+  ```
+  In the above example, the share manager pods will tolerate the `nodetype=storage:NoSchedule` taint on nodes, allowing them to be scheduled on those nodes.
+
+  By combining these three configurations, you can precisely control the data locality of RWX volumes based on your requirements, ensuring optimal performance, resilience, and adherence to organizational policies or constraints.
```

Comment:
Please remove this. I moved the information to the introductory paragraph.

Reply:
Removed. Thank you.